test/install_test.go

```diff
@@ -217,9 +219,32 @@ func TestInstallOrUpgradeCli(t *testing.T) {
 			"--controller-log-level", "debug",
 			"--proxy-log-level", "warn,linkerd2_proxy=debug",
 			"--proxy-version", TestHelper.GetVersion(),
+			fmt.Sprintf("--identity-external-issuer=%s", strconv.FormatBool(TestHelper.ExternalIssuer())),
 		}
 	)
 
+	if TestHelper.ExternalIssuer() {
+
+		// short cert lifetime to put some pressure on the CSR request, response code path
+		args = append(args, "--identity-issuance-lifetime=15s")
+
+		err := TestHelper.CreateNamespaceIfNotExists(TestHelper.GetLinkerdNamespace(), nil)
```

Comment:
`TestHelper.CreateNamespaceIfNotExists` is normally used for creating data plane namespaces because it adds the annotation `linkerd.io/is-test-data-plane: "true"`, used in `bin/test-cleanup`. So the `l5d-integration-external-issuer` namespaces is ending up with this annotation and `linkerd.io/is-control-plane: "true"` as well, which results in an error when invoking `/bin/test-cleanup`.

One option would bet to rename `CreateNamespaceIfNotExists` into a more explicit `CreateDataPlaneNSIfNotExists` function, and create a new `CreateControlPlaneNSIfNotExists` both leveraging some common code.